src/parser/core/Entity.ts

```diff
@@ -211,27 +211,32 @@ class Entity {
     sourceId: number | null = null,
   ): Map<number, number> {
     const buffTimesAtApplication: Map<number, number> = new Map<number, number>();
-    const sortedApplicationTimestamps: number[] = this.getBuffHistory(spellId, sourceId)
-      .map((buffEvent) => buffEvent.timestamp)
-      .sort((a, b) => a - b);
+    const sortedApplicationTimestamps: number[] = [];
+
+    this.getBuffHistory(spellId, sourceId).forEach((e) => {
+      sortedApplicationTimestamps.push(e.timestamp, ...e.refreshHistory);
```

Comment:
'e.refreshHistory' here is an array of timestamps when the buff was refreshed. Since it seems that the intention of the method is to list duration times at all refreshes (I read "re-application" as refresh, but that might be wrong), then the loop is no longer a one-to-one, but one-to-potentially-many. 

Is there is a way to use .map, but potentially insert 1-N elements into the list?

Reply:
Oh, that's `Array.prototype.flatMap<T, U>((t: T) => U[]): U[]`.  It'll map each element to an array of elements and then flatten them all into one array.